### PR TITLE
[Common] utils to handle flex tensor

### PR DESCRIPTION
**Pull request by jaeyun-jung**
add util functions to handle flexible tensor stream.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
This PR is common util functions to handle flexible tensor in nnstreamer element.
See another PRs (#3213, #3216 and #3239) for the usage.

File: gst/nnstreamer/tensor_common.c [hunk 1]
```diff
@@ -336,11 +386,19 @@ gboolean
...
   g_return_val_if_fail (i1 != NULL, FALSE);
   g_return_val_if_fail (i2 != NULL, FALSE);
 
-  if (i1->num_tensors != i2->num_tensors || i1->num_tensors < 1) {
+  flexible = gst_tensors_info_is_flexible (i1);
+  if (flexible != gst_tensors_info_is_flexible (i2)) {
+    return FALSE;
+  }
+
+  compatible = (i1->num_tensors > 0 || i2->num_tensors > 0) || flexible;
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_common.c:399]: Same result, but may be better for readability.
```suggestion
  compatible = (i1->num_tensors > 0 && i2->num_tensors > 0) || flexible;
```

**Review by dongju-chae — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_common.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_common.c]: updated

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**